client/gutenberg/editor/controller.js

```diff
@@ -164,38 +135,6 @@ export const authenticate = ( context, next ) => {
 	}
 };
 
-export const redirect = async ( context, next ) => {
```

Comment:
I was also looking at [removing this code](https://github.com/Automattic/wp-calypso/pull/45938/files#r497586206), but now I'm wondering if it's still needed to [check if the editor flow](https://github.com/Automattic/wp-calypso/blob/master/client/state/selectors/get-selected-editor.js) is forced classic, redirected to wp-admin, or Calypsoified.

Reply:
that's a very good point. We need revert removing this function, and add it as middleware of the routes.

Reply:
We should also add the `waitForSiteIdAndSelectedEditor` logic back then. Otherwise the selected editor state could be potentially not set when visiting directly a `/post` or `/block-editor` URL (the selected editor state is set [here](https://github.com/Automattic/wp-calypso/blob/c8343b8745b408889b8d1384c4f91f7079671b48/client/layout/index.jsx#L183)).